src/main/java/games/strategy/triplea/ai/proAI/ProTechAI.java

```diff
@@ -589,13 +587,12 @@ private static Route getMaxSeaRoute(final GameData data, final Territory start,
   private static List<Territory> getExactNeighbors(final Territory territory, final GameData data) {
     // old functionality retained, i.e. no route condition is imposed.
     // feel free to change, if you are confortable all calls to this function conform.
-    final Match.CompositeBuilder<Territory> endCondBuilder = Match.newCompositeBuilder(
-        Matches.territoryIsImpassable().invert());
-    if (Properties.getNeutralsImpassable(data)) {
-      endCondBuilder.add(Matches.territoryIsNeutralButNotWater().invert());
-    }
+    final Predicate<Territory> endCond = Matches.territoryIsImpassable().invert()
+        .and(Properties.getNeutralsImpassable(data)
+            ? Matches.territoryIsNeutralButNotWater().invert()
+            : Matches.always());
```

Comment:
Could this be changed to use new builder methods?

Reply:
I almost expected to miss something... will fix